src/hssm/plotting/utils.py

```diff
@@ -200,3 +239,154 @@ def _get_title(cols: Iterable["str"], col_values: Iterable[Any]) -> str:
     )
 
     return title
+
+
+def _process_df_for_qp_plot(
+    df: pd.DataFrame, q: int | Iterable[float], cond: str, correct: str | None = None
+) -> pd.DataFrame:
+    """Process the data frame fo the quantile probability plot.
+
+    Parameters
+    ----------
+    df
+        The DataFrame to process.
+    q
+        If an `int`, quantiles will be determined using np.linspace(0, 1, q) (0 and 1
+        will be excluded. If an iterable, will generate quantiles according to this
+        iterable.
+    cond
+        The variable for the conditions.
+    correct
+        The column for whether the answer is correct.
+
+    Returns
+    -------
+    pd.DataFrame
+        The dataframe with `quantile`, `rt`, and `cond` variables.
+    """
+    if isinstance(q, int):
+        q = np.linspace(0, 1, q)[1:-1]
+
```

Comment:
consider adding a check here that checks if min(q) >= 0 and max(q) <= 1.
potentially also check if len(q) < 10 or if int q < 10 --> warning that this is a strange choice and that usually one would pick 4-5 quantiles.

Reply:
Sounds good!